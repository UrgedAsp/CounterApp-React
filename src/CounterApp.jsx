import { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState(value);

  const sumar = (e) => {
    // console.log(e);
    // setCounter( counter+1 );
    setCounter( (c) => c + 1);
  };

  const restar = () => {
    setCounter( counter - 1 );
  };

  const reset = () => {
    setCounter( value );
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{ counter }</h2>
      <button onClick={ sumar }>+1</button>
      <button onClick={ restar }>-1</button>
      <button aria-label="btn-reset" onClick={ reset }>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
