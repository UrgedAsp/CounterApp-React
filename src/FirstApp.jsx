import PropTypes from "prop-types";

// const getSaludo = (nombre)=>{
//   return nombre;
// }

// const newMessage = {Juan : 'Pedro'};

export const FirstApp = ({ title, subtitle = "La tierra explota!", name }) => {
  // console.log(props)

  return (
    <>
      <h1 data-testid="test-title"> {title} </h1>
      {/* <h1>{ getSaludo('Sebastian') }</h1> */}
      {/* <code>{ JSON.stringify(newMessage) }</code> */}
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
  name: "Sebastian Castillo",
  subtitle: "No hay subtitulo",
  title: "No hay t√≠tulo",
};
