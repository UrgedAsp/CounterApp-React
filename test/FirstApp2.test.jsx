import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en FirstApp", () => {
  const title = "Soy un titulo";
  const subtitle = 123;

  test("debe hacer match con el snapshot", () => {
    const { container } = render(
      <FirstApp title={title} subtitle={subtitle} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el titulo", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Debe de mostrar el titulo en un h1", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("Debe de mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
