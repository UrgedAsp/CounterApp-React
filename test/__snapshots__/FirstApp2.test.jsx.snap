// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Soy un titulo
     
  </h1>
  <p>
    123
  </p>
  <p>
    123
  </p>
  <p>
    Sebastian Castillo
  </p>
</div>
`;
